test/hotspot/jtreg/compiler/loopopts/TestLostDependencyOnZeroTripGuard.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright (c) 2021, Red Hat, Inc. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/**
+ * @test
+ * @bug 8263971
+ * @summary C2 crashes with SIGFPE with -XX:+StressGCM and -XX:+StressIGVN
+ *
+ * @run main/othervm -Xcomp -XX:-TieredCompilation -XX:CompileOnly=TestLostDependencyOnZeroTripGuard -XX:+StressGCM -XX:StressSeed=886771365 TestLostDependencyOnZeroTripGuard
```

Comment:
`-XX:+UnlockDiagnosticVMOptions -XX:+IgnoreUnrecognizedVMOptions` should be added for the test, right?

Reply:
`-XX:+UnlockDiagnosticVMOptions` should be enough.

Reply:
Thanks for looking at this

> `-XX:+UnlockDiagnosticVMOptions -XX:+IgnoreUnrecognizedVMOptions` should be added for the test, right?

I added -XX:+UnlockDiagnosticVMOptions

Reply:
I don't think `-XX:+UnlockDiagnosticVMOptions` is enough since `StressGCM` is a C2-only flag.

What will happen if there is no c2 compiler?
Thanks.

Reply:
> I don't think `-XX:+UnlockDiagnosticVMOptions` is enough since `StressGCM` is a C2-only flag.
> 
> What will happen if there is no c2 compiler?

Right. Command line option added.

Reply:
> > I don't think `-XX:+UnlockDiagnosticVMOptions` is enough since `StressGCM` is a C2-only flag.
> > What will happen if there is no c2 compiler?
> 
> Right. Command line option added.

Thanks for your update.

Reply:
> What will happen if there is no c2 compiler?

Ah, yes, good point! I was only thinking about product builds. 

There are actually more compiler tests that have `-XX:+IgnoreUnrecognizedVMOptions` missing as for example: [Vec_mulAddS2I](https://github.com/openjdk/jdk/blob/master/test/hotspot/jtreg/compiler/loopopts/superword/Vec_MulAddS2I.java) that uses C2-specific flags. I'll file a bug to clean that up separately.